test/cuda.jl

```diff
@@ -24,6 +24,11 @@ end
   g_gpu = gradient(x -> w(x), a_gpu)[1]
   @test g_gpu isa CuArray
   @test g_gpu |> collect ≈ g
+
+  # https://github.com/FluxML/Zygote.jl/issues/1027
+  @test gradient(x -> sum(x .!= 0), a_gpu) == (nothing,)
+  g3 = gradient(x -> sum(x .^ 3) / count(x .> 3), a)[1]
+  @test cu(g3) ≈ gradient(x -> sum(x .^ 3) / sum(x .> 3), a_gpu)[1]
```

Comment:
Besides marking count as non-differentiable, the change to `broadcast_forward` removes an error from boolean-valued broadcasts like `x .> 3`.